src/test/app/Offer_test.cpp

```diff
@@ -795,22 +850,831 @@ class Offer_test : public beast::unit_test::suite
             owners ("eve", 1));
     }
 
+    void
+    testcaseSelfCross()
+    {
+        testcase ("Self-crossing");
+
+        using namespace jtx;
+
+        auto const gw = Account {"gateway"};
+        auto const USD = gw["USD"];
+        auto const BTC = gw["BTC"];
+
+        Env env {*this};
+        env.fund (XRP (10000), gw);
+        env (offer (gw, BTC (500), USD (100)));
+        env (offer (gw, USD (100), BTC (500)));
+        env.close ();
+    }
+
+    void
+    testcaseNegativeBalance()
```

Comment:
This is a long enough test that I think it would be nice if it included a narrative about what's getting tested and how.  I suspect the npm test didn't include such comments, but you may have developed an inkling of the intent while you were porting.  It would be nice to capture whatever you figured out in a comment.